src/ClientData/TimerData.cpp

```diff
@@ -63,6 +65,39 @@ std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimers(uint64_t
   return timers;
 }
 
+std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimersAtDepthDiscretized(
+    uint32_t depth, uint32_t resolution, uint64_t start_ns, uint64_t end_ns) const {
+  ORBIT_SCOPE_WITH_COLOR("GetTimersAtDepthDiscretized", kOrbitColorBlueGrey);
+  absl::MutexLock lock(&mutex_);
+  // The query is for the interval [start_ns, end_ns], but it's easier to work with the close-open
+  // interval [start_ns, end_ns+1). We have to be careful with overflowing if end_ns is the maximum
+  // unsigned value. In that case, we will just ignore this max_timestamp for simplicity.
+  end_ns = std::max(end_ns, end_ns + 1);
+
+  if (timers_.find(depth) == timers_.end()) return {};
+
+  std::vector<const orbit_client_protos::TimerInfo*> discretized_timers;
+  uint64_t next_pixel_start_ns = start_ns;
+
+  // We are iterating through all blocks until we are after end_ns, but we don't have other option
+  // if we use TimerChain.
```

Comment:
I would completely remove the comment, or at least the second part of it.

Reply:
Removed the second part.